tests/lib/c_lib/strerror/src/main.c

```diff
@@ -4,9 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#ifdef CONFIG_NEWLIB_LIBC
-#define _POSIX_C_SOURCE 200809
-#endif
+#undef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L /* Needed for strerror() which is tested here */
```

Comment:
Given that the test, issue and workaround existed before this PR, I think it would be best to have this discussion happen outside of this PR and any possible fix be done in another context.

Reply:
Same as https://github.com/zephyrproject-rtos/zephyr/pull/67052#discussion_r1466024433

Reply:
The test was removed in https://github.com/zephyrproject-rtos/zephyr/pull/67891 . Therefore setting this macro is not necessary anymore.